cmd/flightctl/main.go

```diff
@@ -554,23 +558,31 @@ func applyFromReader(client *apiclient.ClientWithResponses, filename string, r i
 				continue
 			}
 			fmt.Printf("%s\n", response.HTTPResponse.Status)
+			httpResponse = *response.HTTPResponse
 		case "Repository":
 			response, err := client.ReplaceRepositoryWithBodyWithResponse(context.Background(), resourceName, "application/json", bytes.NewReader(buf))
 			if err != nil {
 				errs = append(errs, err)
 				continue
 			}
 			fmt.Printf("%s\n", response.HTTPResponse.Status)
+			httpResponse = *response.HTTPResponse
 		case "ResourceSync":
 			response, err := client.ReplaceResourceSyncWithBodyWithResponse(context.Background(), resourceName, "application/json", bytes.NewReader(buf))
 			if err != nil {
 				errs = append(errs, err)
 				continue
 			}
 			fmt.Printf("%s\n", response.HTTPResponse.Status)
+			httpResponse = *response.HTTPResponse
 		default:
 			errs = append(errs, fmt.Errorf("%s: skipping resource of unkown kind %q: %v", filename, kind, resource))
 		}
+
+		// bad HTTP Responses don't generate an error on the OpenAPI client, we need to check the status code manually
+		if httpResponse.StatusCode != http.StatusOK && httpResponse.StatusCode != http.StatusCreated {
```

Comment:
if we hit the default case would this panic?

Reply:
no it will just be status 0 I thought it was a ref my bad :)

Reply:
ah but it would emit an additional error in such case. I'll fix that

Reply:
done